components/org.wso2.carbon.identity.query.saml/src/test/java/org/wso2/carbon/identity/query/saml/validation/SAMLAttributeQueryValidatorTest.java

```diff
@@ -0,0 +1,124 @@
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.wso2.carbon.identity.query.saml.validation;
+
+import org.mockito.Mock;
+import org.opensaml.saml.common.SAMLVersion;
+import org.opensaml.saml.saml2.core.NameID;
+import org.opensaml.saml.saml2.core.RequestAbstractType;
+import org.opensaml.saml.saml2.core.Subject;
+import org.opensaml.saml.saml2.core.impl.IssuerImpl;
+import org.opensaml.saml.saml2.core.impl.NameIDImpl;
+import org.opensaml.saml.saml2.core.impl.SubjectImpl;
+import org.opensaml.saml.saml2.core.impl.SubjectQueryImpl;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.testng.PowerMockTestCase;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.base.CarbonBaseConstants;
+import org.wso2.carbon.identity.core.model.SAMLSSOServiceProviderDO;
+import org.wso2.carbon.identity.query.saml.dto.InvalidItemDTO;
+import org.wso2.carbon.identity.query.saml.exception.IdentitySAML2QueryException;
+import org.wso2.carbon.identity.query.saml.internal.SAMLQueryServiceComponent;
+import org.wso2.carbon.identity.query.saml.util.OpenSAML3Util;
+import org.wso2.carbon.identity.query.saml.util.SAMLQueryRequestConstants;
+import org.wso2.carbon.identity.query.saml.util.SAMLQueryRequestUtil;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.UserRealm;
+import org.wso2.carbon.user.core.UserStoreManager;
+import org.wso2.carbon.user.core.service.RealmService;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
+
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Matchers.anyString;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.wso2.carbon.identity.query.saml.validation.TestUtil.*;
+
+import static org.testng.Assert.*;
+
+/**
+ * Test Class for the SAMLAttributeQueryValidator
+ */
+@PrepareForTest({MultitenantUtils.class, SAMLQueryServiceComponent.class, SAMLQueryRequestUtil.class, OpenSAML3Util.class})
+public class SAMLAttributeQueryValidatorTest extends PowerMockTestCase {
+
+    @Mock
+    RealmService testRealmService;
+    @Mock
+    UserRealm testUserRealm;
+    @Mock
+    UserStoreManager testuserStoreManager;
+
+    SAMLAttributeQueryValidator testclass = new SAMLAttributeQueryValidator();
+    List<InvalidItemDTO> invalidItems = new ArrayList<>();
+
+    @BeforeClass
+    public void setUp() {
+
+        initPrivilegedCarbonContext("testDomain", 1, "testuser");
+    }
+
+    @AfterClass
+    public void tearDown() {
+
+        System.clearProperty(CarbonBaseConstants.CARBON_HOME);
```

Comment:
Shouldn't you end the tenant flow in the tearDown?

Reply:
ya